src/maliput_viz/plugins/maliput_mesh_converter.cc

```diff
@@ -0,0 +1,189 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Note: code in this file was forked and changed from this file:
+// https://github.com/RobotLocomotion/drake/blob/master/automotive/maliput/utility/generate_obj.cc
+// and this commit: 82bf3c8a02678f553c746bdbbe0f8e5a345841b7
+
+#include "maliput_mesh_converter.hh"
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <fstream>
+#include <functional>
+#include <initializer_list>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+#include <ignition/common/SubMesh.hh>
+#include <ignition/math/Vector3.hh>
+#include <maliput/api/branch_point.h>
+#include <maliput/api/junction.h>
+#include <maliput/api/lane.h>
+#include <maliput/api/lane_data.h>
+#include <maliput/api/road_geometry.h>
+#include <maliput/api/segment.h>
+#include <maliput/common/maliput_abort.h>
+#include <maliput/utility/mesh.h>
+
+using namespace maliput::utility::mesh;
+
+namespace maliput {
+namespace viz {
+namespace mesh {
+namespace {
+
+// Sorts a vector of tuples composed of 3D points and an index in a polar and
+// counterclockwise way.
+//
+// Points in `unordered_vector` are supposed to be on the same plane. Based on
+// that, the mid point of all of them is computed and then used as a center of
+// coordinates. Angles are computed using that center and the compared to sort
+// the vector.
+// @returns A std::vector<std::tuple<ignition::math::Vector3d, int>> with the
+// same items of `unordered_vector` but sorted in a polar and counterclockwise
+// order.
+// @throws std::runtime_error When `unordered_vector`' size is zero.
+std::vector<std::tuple<ignition::math::Vector3d, int>> PolarSort(
+    const std::vector<std::tuple<ignition::math::Vector3d, int>>& unordered_vector) {
+  MALIPUT_DEMAND(unordered_vector.size() > 0);
+
+  std::vector<std::tuple<ignition::math::Vector3d, int>> ordered_vector = unordered_vector;
+
+  // Computes the center.
+  std::vector<ignition::math::Vector3d> points;
+  ignition::math::Vector3d center;
+  for (const std::tuple<ignition::math::Vector3d, int>& vertex_index : unordered_vector) {
+    center += std::get<0>(vertex_index);
+  }
+  center /= static_cast<double>(points.size());
+
+  // TODO(agalbachicar)    This polar sort should be done with respect to the
```

Comment:
Shall we create / migrate a ticket for this TODO?  Is it still relevant?

Reply:
Done. https://github.com/maliput/maliput_viz/issues/4